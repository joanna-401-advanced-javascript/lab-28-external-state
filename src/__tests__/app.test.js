import React from 'react';
import renderer from 'react-test-renderer';
import App from '../app';

describe('<App />', () => {
  it('nothing from this page is rendered at application start', () => {
    const app = shallow(<App />);
    expect(app.find('div').exists()).toBe(true);
  });

  it('state contains count', () => {
    const app = mount(<App />);
    expect(app.state('count')).toEqual(0);
  });

  it('state contains name', () => {
    const app = mount(<App />);
    expect(app.state('name')).toEqual(null);
  });

  it('snapshot is rendered correctly', () => {
    const app = renderer
      .create(<App />)
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
