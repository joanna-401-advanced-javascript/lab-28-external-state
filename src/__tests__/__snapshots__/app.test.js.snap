// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot is rendered correctly 1`] = `
<div>
  <div>
    Name: 
  </div>
  <div>
    # of Updates: 
    0
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
    />
  </form>
</div>
`;
